azurerm/resource_arm_powerbidedicated_capacity.go

```diff
@@ -0,0 +1,233 @@
+		return fmt.Errorf("Error creating Capacity %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for creation of Capacity %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	resp, err := client.GetDetails(ctx, resourceGroup, name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving Capacity %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	if resp.ID == nil || *resp.ID == "" {
+		return fmt.Errorf("Cannot read Capacity %q (Resource Group %q) ID", name, resourceGroup)
+	}
+	d.SetId(*resp.ID)
+
+	return resourceArmPowerBIDedicatedCapacityRead(d, meta)
+}
+
+func resourceArmPowerBIDedicatedCapacityRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).PowerBIDedicated.CapacityClient
+	ctx, cancel := timeouts.ForRead(meta.(*ArmClient).StopContext, d)
+	defer cancel()
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["capacities"]
+
+	resp, err := client.GetDetails(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Capacity %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading Capacity %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.Set("name", name)
+	d.Set("resource_group_name", resourceGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+	if props := resp.DedicatedCapacityProperties; props != nil {
+		if err := d.Set("administrators", utils.FlattenStringSlice(props.Administration.Members)); err != nil {
+			return fmt.Errorf("Error setting `administration`: %+v", err)
+		}
+	}
+	if err := d.Set("sku", resp.Sku.Name); err != nil {
+		return fmt.Errorf("Error setting `sku`: %+v", err)
+	}
+
+	return tags.FlattenAndSet(d, resp.Tags)
+}
+
+func resourceArmPowerBIDedicatedCapacityUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).PowerBIDedicated.CapacityClient
+	ctx, cancel := timeouts.ForUpdate(meta.(*ArmClient).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	administrators := d.Get("administrators").(*schema.Set).List()
+	sku := d.Get("sku").(string)
+	t := d.Get("tags").(map[string]interface{})
+
+	parameters := powerbidedicated.DedicatedCapacityUpdateParameters{
+		DedicatedCapacityMutableProperties: &powerbidedicated.DedicatedCapacityMutableProperties{
+			Administration: &powerbidedicated.DedicatedCapacityAdministrators{
+				Members: utils.ExpandStringSlice(administrators),
+			},
+		},
+		Sku: &powerbidedicated.ResourceSku{
+			Name: utils.String(sku),
+		},
+		Tags: tags.Expand(t),
+	}
+
+	future, err := client.Update(ctx, resourceGroup, name, parameters)
```

Comment:
since there's a separate update function presumably this supports delta updates?

Reply:
Yes. I think you should be correct since it can update property "sku", "administrators" and "tags" successfully without attribute "ForceNew".